code/game/machinery/antiquemattersynth.dm

```diff
@@ -0,0 +1,163 @@
+list("category" = "tools", "name" = "flare gun", "path" =/obj/item/weapon/gun/projectile/flare, cost = 2000),
+
+list("category" = "machinery", "name" = "air pump", "path" = /obj/machinery/portable_atmospherics/pump, "cost" = 500),
+list("category" = "machinery", "name" = "portable scrubber", "path" = /obj/machinery/portable_atmospherics/scrubber, "cost" = 500),
+list("category" = "machinery", "name" = "space heater", "path" = /obj/machinery/space_heater, "cost" = 500),
+list("category" = "machinery", "name" = "air conditioner", "path" = /obj/machinery/space_heater/air_conditioner, "cost" = 500),
+list("category" = "machinery", "name" = "cart", "path" = /obj/machinery/cart/cargo, "cost" = 200),
+list("category" = "machinery", "name" = "tractor", "path" = /obj/structure/bed/chair/vehicle/tractor, "cost" = 2000),
+list("category" = "machinery", "name" = "rad collector", "path" = /obj/machinery/power/rad_collector, "cost" = 1000),
+list("category" = "machinery", "name" = "emitter", "path" = /obj/machinery/power/emitter, "cost" = 1000),
+list("category" = "machinery", "name" = "prism", "path" = /obj/machinery/prism, "cost" = 2000),
+list("category" = "machinery", "name" = "MSGS", "path" = /obj/machinery/atmospherics/binary/msgs/unanchored, "cost" = 2000),
+)
+
+/obj/machinery/power/antiquesynth
+	name = "antique matter synthesizer"
+	desc = "An ancient piece of salvaged tech from a period before matter synthesizers were small enough to be moved around. This one is designed with industrial purposes in mind and consumes huge amounts of power."
+
+	icon = 'icons/obj/xenoarchaeology.dmi'
+	icon_state = "scanner_0old"
+
+	use_power = 1
+	density = 1
+	anchored = 0
+	machine_flags = WRENCHMOVE | FIXED2WORK | EMAGGABLE
+	req_access = list(access_engine)
+
+	var/consumption = 0 //How much are we set to draw off the net? Clamped between 0 and 2 GIGAWATT (2,000,000,000 Watts)
+	var/on = 0
+	var/charge = 0 //How much we've stored. Also capped at 2 GIGAWATT.
+	var/category = "resources" //which list to display
+	var/list/categories = list(list("category" = "resources"), list("category" = "tools"), list("category" = "machinery")) //Yes it is necessary to write the list like this
+
+/obj/machinery/power/antiquesynth/proc/toggle_power()
+	on = !on
+	if(!get_powernet())
+		on = FALSE
+		visible_message("<span class='warning'>The [src] buzzes and shuts off.</span>")
+	update_icon()
+
+/obj/machinery/power/antiquesynth/update_icon()
+	return
+	//Maybe I'll add more?
+
+/obj/machinery/power/antiquesynth/process()
+	if(!on)
+		return
+	if(!anchored || !get_powernet())
+		toggle_power()
+		return
+	if(charge >= GIGAWATT)
+		charge = min(charge, GIGAWATT)
+		return //We can't get more charged than this!
+	if(avail()>consumption)
+		charge += consumption
+		add_load(consumption)
+
+/obj/machinery/power/antiquesynth/attack_ai(var/mob/user as mob)
+	to_chat(user, "<span class='warning'>You aren't equipped to interface with technology this old!</span>")
+	return 0
```

Comment:
there's no code that expects a return value from `attack_ai`, you can safely remove this line